Sources/PackageGraph/Pubgrub.swift

```diff
@@ -516,6 +594,30 @@ public final class PubgrubDependencyResolver<
     /// Skip updating containers while fetching them.
     private let skipUpdate: Bool
 
+    func trace(
+        value: Traceable,
+        type: TraceStep.StepType,
+        location: TraceStep.Location,
+        cause: String?
+        ) {
+        delegate?.trace(TraceStep(value: value,
+                                  type: type,
+                                  location: location,
+                                  cause: cause,
+                                  decisionLevel: solution.decisionLevel))
+    }
+
+    func decide(_ package: Identifier, version: Version, location: TraceStep.Location) {
+        let term = Term(package, .versionSet(.exact(version)))
+        trace(value: term, type: .decision, location: location, cause: nil)
+        solution.decide(term)
+    }
+
+    func derive(_ term: Term<Identifier>, cause: Incompatibility<Identifier>, location: TraceStep.Location) {
```

Comment:
Not sure if it's worth to wrap these two calls in a separate method

Reply:
I just wanted the call to `trace()` to be in there so that I don't miss it. But I do agree that the extra method for that feels overkill.